### PR TITLE
feat(search): present view for errors

### PR DESCRIPTION
**Pull request by cyndichin**
## Summary
Present view for errors from online search. 

## References 
IN-1110

## Implementation Details
Added a new `EmptyStateViewModel` called `ErrorEmptyState` to handle errors returned from our Online Search for Archive and All Items. Added new `bannerData` to our `SearchViewModel` which is presented when we set `showBanner` to true when an Online Search for Saves fails. This banner is presented for offline state and unhandled error states.

## Test Steps
For saves search:
- [ ] When I submit a search query 
AND the query fails  
AND the failure is an unhandled error
THEN perform local search
AND display banner notifying the user something went wrong

For archive search:
- [ ] When I submit a search query 
AND the query fails  
AND the failure is an unhandled error
THEN present an error empty state to communicate the error

For all items search:
- [ ] When I submit a search query 
AND the query fails  
AND the failure is an unhandled error
THEN present an error empty state to communicate the error
Designs: See gig attached. 

## PR Checklist:
- [x] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

## Screenshots

|Archive / All Items Error State|Saves Error State|
|------|-----|
|Archive / All Items Error State|Saves Error State|

### CODE REVIEW

File: PocketKit/Sources/PocketKit/MyList/EmptyStates/Search/ErrorEmptyState.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import Foundation
+import Textile
+
+struct ErrorEmptyState: EmptyStateViewModel {
+    let imageAsset: ImageAsset = .warning
+    let maxWidth: CGFloat = Width.normal.rawValue
+    let icon: ImageAsset? = nil
+    let headline: String? = L10n.oopsTryAgain
+    let detailText: String? = L10n.WeReExperiencingAnErrorAndTemporarilyCanTGetYouWhatYouNeed.pleaseTryAgainLater
```
Comment (cyndichin) [inline PocketKit/Sources/PocketKit/MyList/EmptyStates/Search/ErrorEmptyState.swift:13]: our localization variable names are a bit long. However, I decided to use the current implementation as opposed to refactoring it as part of this work. We should discuss on what our stance is on this. Do we continue with this for now?

Comment (cyndichin) [reply PocketKit/Sources/PocketKit/MyList/EmptyStates/Search/ErrorEmptyState.swift:13]: Seems like we'll look at it as a followup - https://pocket.slack.com/archives/C01UH57EJKD/p1676391105038829?thread_ts=1676333516.460409&cid=C01UH57EJKD

Comment (bassrock) [reply PocketKit/Sources/PocketKit/MyList/EmptyStates/Search/ErrorEmptyState.swift:13]: This is what @Gio2018 did https://github.com/Pocket/pocket-ios/pull/402/files#diff-adf47436386e17ed486299bf385bee0272f10cf98b3a79f17cbff3995cecdff4R141

**Review by bassrock — APPROVED** [review]
Yay! Feel free to update the string keys as you see fit.

File: PocketKit/Sources/PocketKit/MyList/EmptyStates/Search/ErrorEmptyState.swift (hunk 1 above)
Comment (cyndichin) [reply PocketKit/Sources/PocketKit/MyList/EmptyStates/Search/ErrorEmptyState.swift]: that's awesome! okay updated the keys, will merge after successful build

File: PocketKit/Sources/PocketKit/Resources/en.lproj/Localizable.strings [hunk 2]
```diff
@@ -128,3 +128,10 @@
 "It's rude, vulgar, or offensive" = "It's rude, vulgar, or offensive";
 "It contains misinformation" = "It contains misinformation";
 "Other" = "Other";
+
+// Search
+"search.errorHeadline" = "Oops! Try again?";
+"search.errorMessage" = "We’re experiencing an error and temporarily can’t get you what you need. Please try again later.";
+"search.limitedResults" = "Limited search results";
+"search.offlineMessage" = "You can only search titles and URLs while offline. Connect to the internet to use Premium's full-text search.";
+"search.banner.errorMessage." = "We’re experiencing an error and can’t show you full search results. Please try again later.";
```
Comment (Gio2018) [inline PocketKit/Sources/PocketKit/Resources/en.lproj/Localizable.strings:137]: 🎉

**Review by Gio2018 — APPROVED** [review]
Awesome @cyndichin ! Tried what we just discussed on Zoom and works as expected!
:shipit:

File: PocketKit/Sources/PocketKit/Resources/en.lproj/Localizable.strings (hunk 2 above)
Comment (cyndichin) [reply PocketKit/Sources/PocketKit/Resources/en.lproj/Localizable.strings:137]: > Awesome @cyndichin ! Tried what we just discussed on Zoom and works as expected!
> :shipit:

thank you!

**Pull request merged**